### ISSUE TITLE
model question!

### ISSUE THREAD

**Issue by yujack333** [tkipf/gcn#133]
Hi,  i have a question about the model.
As far as i know, the loss of the auto-encoder model want the inputs and outputs of the model are as identical as possible. And the inputs of the model is train graph Adjacency matrix. So i want ask how can the model predict the edges in test graph if the model trained well(during training, the model see the edges in test graph as negative).  
Looking forward to your reply. Best wishes for you.

**Comment by tkipf** [comment]
This is an excellent question. In fact, the model will need a certain
inductive bias to generalize well to the test set, which in this case
requires some degree of regularization. This can be achieved by monitoring
performance on a validation set and use, e.g., early stopping.

**Comment by yujack333** [comment]
Thank you for your answer, it help me a lot.

**Comment by QasimWani** [comment]
@tkipf what if the graph is the same for train and test and the only thing that changes is temporal information, i.e. features per node. Is GCN not the best solution in that case, or is a simple auto-encoder structure better and GCN an overkill?
Thanks
